src/System.Net.HttpListener/src/System/Net/Windows/RequestContextBase.cs

```diff
@@ -116,7 +125,19 @@ protected void UnsetBlob()
 
         protected void SetBuffer(int size)
         {
-            _backingBuffer = size == 0 ? null : new byte[size];
+            if(_backingBuffer != IntPtr.Zero)
+            {
+                Marshal.FreeHGlobal(_backingBuffer);
+            }
+
+            _backingBuffer = size == 0 ? IntPtr.Zero : Marshal.AllocHGlobal(size);
+            _backingBufferLength = size;
+
+            // Zero out the contents of the buffer.
+            for(int i = 0; i < size; ++i)
+            {
+                Marshal.WriteByte(_backingBuffer + i, 0);
+            }
```

Comment:
How big are these usually?  Calling Marshal.WriteByte per byte for a large buffer isn't cheap.  Alternatively you could use span:
```C#
new Span<byte>((byte*)_backingBuffer, 0, size).Fill(0);
```

Reply:
The default size is 4096, but they can be larger than that depending on the size of the request. I really don't like having that loop there, so I'll look into doing it that way instead. Thanks!